### PR TITLE
fix: validate regex patterns in annotations #510

### PR DESCRIPTION
**Pull request by Pandry**
Hiya!
I'm writing a PR for the issue #510 
I managed to write some code to fix the missing check
Please do not merge yet, as I still need to write the test

### CODE REVIEW

File: pkg/webhook/tenant/forbidden_annotations_regex.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//nolint:dupl
```
Comment (prometherion) [inline pkg/webhook/tenant/forbidden_annotations_regex.go:3]: I guess there's no issue with `dupl` linter here, let's try to remove and run locally the linters to check if this is really required.

**Review by ptx96 — DISMISSED** [review]
Hi @Pandry ;
thank you so much for your contribution.

Could you please translate [this](https://github.com/clastix/capsule/pull/513/commits/5bc2091905e11b142cfa056da7261169090c3300) commit message to english?

**Comment by Pandry** [comment]
> Could you please translate [this](https://github.com/clastix/capsule/pull/513/commits/5bc2091905e11b142cfa056da7261169090c3300) commit message to english?

Dang! Sorry about that! Fixed!

File: e2e/forbidden_annotations_regex_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		}
+	}
+
+	successRegexes := []string{
+		"",
+		"(.*gitops|.*nsm)",
+	}
+	for _, annotation := range annotationsToCheck {
+		for _, annotationValue := range successRegexes {
+			It("should suceed using a value non matching the regex", func() {
```
Comment (prometherion) [inline e2e/forbidden_annotations_regex_test.go:77]: Put the value of the annotation in the title of the feature test: using the `--focus` Ginkgo CLI flag we can filter tests by their name.

**Review by prometherion — CHANGES_REQUESTED** [review]
Commit fcbdac298bd24a55de2c3f2a2999a71061a0db35 must be renamed in lowercase, suggested:

> tests(e2e): tenant regex forbidden namespace labels and annotations

Same for debf060bccf6d12671d04e2e088e446282f2e97d: since addressing changes in the same PR, please, squash it with 01ac1be6fa844acb94d4ae6c976fd007692ff20d.

File: e2e/forbidden_annotations_regex_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
+//go:build e2e
+// +build e2e
```
[the remark this answers is not in the document]
Comment (Pandry) [reply e2e/forbidden_annotations_regex_test.go:2]: Dagn! Turned out VS Code added it without me noticing!

File: pkg/webhook/tenant/forbidden_annotations_regex.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (prometherion) [inline pkg/webhook/tenant/forbidden_annotations_regex.go:1]: Wrong header here:

```
// Copyright 2020-2021 Clastix Labs
// SPDX-License-Identifier: Apache-2.0
```

Comment (Pandry) [reply pkg/webhook/tenant/forbidden_annotations_regex.go:2]: This didn't seem right to me, but anyway, done

File: e2e/forbidden_annotations_regex_test.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	capsulev1alpha1 "github.com/clastix/capsule/api/v1alpha1"
```
Comment (prometherion) [inline e2e/forbidden_annotations_regex_test.go:15]: Why do we need the old `capsulev1alpha1` import here?

All tests must be addressed on the stored one, `v1beta1`.

**Review by prometherion — APPROVED** [review]

**Review by prometherion — CHANGES_REQUESTED** [review]
Amend commit message from `tests` to `test`

**Comment by Pandry** [comment]
> Amend commit message from `tests` to `test`

Ok, done

**Review by prometherion — APPROVED** [review]
LGTM, thanks a lot for the hard work on this!

**Pull request merged**